ndcube/ndcube.py

```diff
@@ -894,3 +894,223 @@ def to(self, new_unit, **kwargs):
             A new instance with the new unit and data and uncertainties scales accordingly.
         """
         return self * (self.unit.to(new_unit, **kwargs) * new_unit / self.unit)
+
+    def rebin(self, bin_shape, **kwargs):
+        """Downsample array by combining contiguous pixels into bins.
+
+        Values in bins are determined by applying a function to the pixel values within it.
+        The number of pixels in each bin in each dimension is given by the bin_shape input.
+        This must be an integer fraction of the cube's array size in each dimension.
+        If the NDCube instance has uncertainties attached, they are propagated
+        depending on binning method chosen.
+
+        Parameters
+        ----------
+        bin_shape : array-like
+            The number of pixels in a bin in each dimension.
+            Must be the same length as number of dimensions in data.
+            Each element must be in int. If they are not they will be rounded
+            to the nearest int.
+        operation : function
+            Function applied to the data to derive values of the bins.
+            Default is `numpy.mean`
+        use_masked_values: `bool`
+            Determines how masked values are handled.
+            If False (default), masked values are excluded when calculating rebinned value.
+            If True, masked values are used in calculating rebinned value.
+        handle_mask: `None` or function
+            Function to apply to each bin in the mask to calculate the new mask values.
+            If `None` resultant mask is `None`.
+            Default=`numpy.all`
```

Comment:
```suggestion
            Default is `numpy.all`
```
Just to match the default style from `operation`